blogposts/2021/eliminating-secrets-from-codebase.md

```diff
@@ -0,0 +1,134 @@
D%7B35%7D%20or%0A%0A%2F%2F%20Google%20Cloud%20Platform%20API%20Key%0AAIza%5B0-9A-Za-z%5C%5C-_%5D%7B35%7D%20or%0A%0A%2F%2F%20Google%20Cloud%20Platform%20OAuth%0A%5B0-9%5D%2B-%5B0-9A-Za-z_%5D%7B32%7D%5C%5C.apps%5C%5C.googleusercontent%5C%5C.com%20or%0A%0A%2F%2F%20Google%20Drive%20API%20Key%0AAIza%5B0-9A-Za-z%5C%5C-_%5D%7B35%7D%20or%0A%0A%2F%2F%20Google%20Drive%20OAuth%0A%5B0-9%5D%2B-%5B0-9A-Za-z_%5D%7B32%7D%5C%5C.apps%5C%5C.googleusercontent%5C%5C.com%20or%0A%0A%2F%2F%20Google%20(GCP)%20Service-account%0A%5C%22type%5C%22%3A%20%5C%22service_account%5C%22%20or%0A%0A%2F%2F%20Google%20Gmail%20API%20Key%0AAIza%5B0-9A-Za-z%5C%5C-_%5D%7B35%7D%20or%0A%0A%2F%2F%20Google%20Gmail%20OAuth%0A%5B0-9%5D%2B-%5B0-9A-Za-z_%5D%7B32%7D%5C%5C.apps%5C%5C.googleusercontent%5C%5C.com%20or%0A%0A%2F%2F%20Google%20OAuth%20Access%20Token%0Aya29%5C%5C.%5B0-9A-Za-z%5C%5C-_%5D%2B%20or%0A%0A%2F%2F%20Google%20YouTube%20API%20Key%0AAIza%5B0-9A-Za-z%5C%5C-_%5D%7B35%7D%20or%0A%0A%2F%2F%20Google%20YouTube%20OAuth%0A%5B0-9%5D%2B-%5B0-9A-Za-z_%5D%7B32%7D%5C%5C.apps%5C%5C.googleusercontent%5C%5C.com%20or%0A%0A%2F%2F%20Heroku%20API%20Key%0A%5BhH%5D%5BeE%5D%5BrR%5D%5BoO%5D%5BkK%5D%5BuU%5D.*%5B0-9A-F%5D%7B8%7D-%5B0-9A-F%5D%7B4%7D-%5B0-9A-F%5D%7B4%7D-%5B0-9A-F%5D%7B4%7D-%5B0-9A-F%5D%7B12%7D%20or%0A%0A%2F%2F%20MailChimp%20API%20Key%0A%5B0-9a-f%5D%7B32%7D-us%5B0-9%5D%7B1%2C2%7D%20or%0A%0A%2F%2F%20Mailgun%20API%20Key%0Akey-%5B0-9a-zA-Z%5D%7B32%7D%20or%0A%0A%2F%2F%20Password%20in%20URL%0A%5Ba-zA-Z%5D%7B3%2C10%7D%3A%2F%2F%20%5B%5E%2F%5C%5Cs%3A%40%5D%7B3%2C20%7D%3A%5B%5E%2F%5C%5Cs%3A%40%5D%7B3%2C20%7D%40.%7B1%2C100%7D%5B%5C%22%27%5C%5Cs%5D%20or%0A%0A%2F%2F%20PayPal%20Braintree%20Access%20Token%0Aaccess_token%5C%5C%24production%5C%5C%24%5B0-9a-z%5D%7B16%7D%5C%5C%24%5B0-9a-f%5D%7B32%7D%20or%0A%0A%2F%2F%20Picatic%20API%20Key%0Ask_live_%5B0-9a-z%5D%7B32%7D%20or%0A%0A%2F%2F%20Slack%20Webhook%0Ahttps%3A%2F%2F%20hooks%5C%5C.slack%5C%5C.com%2Fservices%2FT%5Ba-zA-Z0-9_%5D%7B8%7D%2FB%5Ba-zA-Z0-9_%5D%7B8%7D%2F%5Ba-zA-Z0-9_%5D%7B24%7D%20or%0A%0A%2F%2F%20Stripe%20API%20Key%0Ask_live_%5B0-9a-zA-Z%5D%7B24%7D%20or%0A%0A%2F%2F%20Stripe%20Restricted%20API%20Key%0Ark_live_%5B0-9a-zA-Z%5D%7B24%7D%20or%0A%0A%2F%2F%20Square%20Access%20Token%0Asq0atp-%5B0-9A-Za-z%5C%5C-_%5D%7B22%7D%20or%0A%0A%2F%2F%20Square%20OAuth%20Secret%0Asq0csp-%5B0-9A-Za-z%5C%5C-_%5D%7B43%7D%20or%0A%0A%2F%2F%20Telegram%20Bot%20API%20Key%0A%5B0-9%5D%2B%3AAA%5B0-9A-Za-z%5C%5C-_%5D%7B33%7D%20or%0A%0A%2F%2F%20Twilio%20API%20Key%0ASK%5B0-9a-fA-F%5D%7B32%7D%20or%0A%0A%2F%2F%20Twitter%20Access%20Token%0A%5BtT%5D%5BwW%5D%5BiI%5D%5BtT%5D%5BtT%5D%5BeE%5D%5BrR%5D.*%5B1-9%5D%5B0-9%5D%2B-%5B0-9a-zA-Z%5D%7B40%7D%20or%0A%0A%2F%2F%20Twitter%20OAuth%0A%5BtT%5D%5BwW%5D%5BiI%5D%5BtT%5D%5BtT%5D%5BeE%5D%5BrR%5D.*%5B%27%7C%5C%22%5D%5B0-9a-zA-Z%5D%7B35%2C44%7D%5B%27%7C%5C%22%5D%22>), then follow along the comments to tune the search for your repositories. You can also [learn more about getting started with Sourcegraph here](https://docs.sourcegraph.com/getting-started).
```

Comment:
FYI the search button on this seems broken to me, it doesn't do anything for me (in Safari 15).

Reply:
I think it works for me?

<img width="1424" alt="Screenshot 2021-09-24 at 15 43 18" src="https://user-images.githubusercontent.com/82806852/134693941-05dc65a3-d03b-46a7-92a9-422b3f966615.png">

Reply:
What happens when you click the "Search" button? Nothing happens for me

Reply:
🤦‍♀️  I see what you mean. No activity on clicking for me either. @andreeleuterio something for you to check please!

Reply:
It doesn't work until you put in your own repo pattern following the top comment. This will be a bit of challenge for readers thinking they can get immediate results without modifying the notebook. The article does tell readers to follow along with the comments, but people skimming might not catch that. Maybe add a working repo pattern there that we know won't expose actual secrets?

Reply:
Yeah admittedly I hadn't read the article. But people's attention span on the internet is short 😄 
I think the issue here is more that there is no error messaging on the website at all. If it showed a notice on click "You need to put in a repo pattern" it would be fine.

Reply:
Great points folks. I think a good solution is to have `repo:sourcegraph/sourcegraph` instead of `[your targeted repos]` but keep the comment above for folks skimming through it. What do you think?

Reply:
I think this works!